### ISSUE TITLE
About data set conversion

### ISSUE THREAD

**Issue by ishakpacal76** [lkeab/BCNet#8]
Hey dear @lkeab 
Thanks for this work. I want to use BCNet, but I only have about 1000 RGB images and their ground truth images. There is no problem with training with object detection such as Faster R-CNN, but when I try to do it with BCNet or mask-rcnn, I get an error "Cannot find field 'gt_masks' in the given Instances". How can I register dataset with ground truth without json segmentation file? In other words, ground truth is sufficient with RGB images to do instance segmentation, or how can I do it.

Thanks a lot in advance.

**Comment by ishakpacal76** [comment]
Hey @lkeab Thanks for quick reply,
First, I have medical images which have rgb images and corresponding ground truth and bounding boxes (xml files).
I want object segmentation (semantic segmentation). So my question is: How can I register data set with ground truth without json segmentation file?, because I dont have json segmentation file (polygon). Do you have any sugesstion to train BCNet.
Thanks in advance.

**Comment by lkeab** [comment]
Hi, For registering dataset, I think the comments in this [file](https://github.com/lkeab/BCNet/blob/main/detectron2/data/datasets/register_coco.py) will help. If you meet the  "Cannot find field 'gt_masks' in the given Instances, you can delete the corresponding code line for requiring this field. Also, you can refer to the code of some pure detection project based on detectron2 (such as this [project](https://github.com/lkeab/gsnet) for car keypoint detection or FCOS). BCNet requires GT mask annotation during training (maybe you can use saliency detection or some well pretrained model to generate pseudo mask label for training in your case.)

**Comment by ishakpacal76** [comment]
It steel needs json annotation, but I have ground truth mask. If you add a script etc. to register dataset with ground truth it will be perfect.

**Comment by lkeab** [comment]
Yes, there should be a script convert your gt mask to coco format. This [code](https://github.com/chrise96/image-to-coco-json-converter) may help.

**Comment by ishakpacal76** [comment]
Thanks, I am working on it. 
The point I'm wondering is, can we do register dataset  for detectron2 with just ground truth masks, without json annotation?
Anyway, thanks again.
